scripts/masterCron.ts

```diff
@@ -211,6 +249,47 @@ async function getAnnualizedVol(underlying: string, resolution: number) {
   return candles[candles.length - 1][pricePoint] * 10 ** 6;
 }
 
+async function updateTokenList(
+  fileName: string,
+  vaultArtifactAbi: any,
+  ierc20Abi: any,
+  provider: any,
+  network: string
+) {
+  let tokens = [];
+
+  let token = {
+    chainId: 1,
+    address: "",
+    name: "",
+    symbol: "",
+    decimals: 0,
+  };
+
+  for (let vaultName in deployments[network].vaults) {
+    const vault = new ethers.Contract(
+      deployments[network].vaults[vaultName].address,
+      vaultArtifactAbi,
+      provider
+    );
+
+    let oToken = new ethers.Contract(
+      await vault.nextOption(),
+      ierc20Artifact.abi,
+      provider
+    );
+
+    token["address"] = oToken.address;
+    token["name"] = await oToken.name();
+    token["symbol"] = await oToken.symbol();
+    token["decimals"] = await oToken.decimals();
+
+    tokens += token;
```

Comment:
This is better re-written by declaring the variable in the loop itself:

```typescript
const token: Token = {
 ...
}
tokens.push(token); // The '+' operator creates weird behaviors
```

Reply:
fixed